site/en/tutorials/load_data/images.ipynb

```diff
@@ -0,0 +1,1356 @@
+        "  start = time.time()\n",
+        "  for i,(images,labels) in enumerate(it):\n",
+        "    if i%10 == 0:\n",
+        "      print('.',end='')\n",
+        "  print()\n",
+        "  end = time.time()\n",
+        "\n",
+        "  duration = end-start\n",
+        "  print(\"{} batches: {} s\".format(batches, duration))\n",
+        "  print(\"{:0.5f} Images/s\".format(BATCH_SIZE*batches/duration))\n",
+        "  print(\"Total time: {}s\".format(end-overall_start))"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "TYiOr4vdLcNX",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "The performance of the current dataset is:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "qwOJGxw_FkJU",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "ds = image_label_ds.repeat()\n",
+        "ds = ds.shuffle(buffer_size=4000) \n",
+        "ds = ds.batch(BATCH_SIZE).prefetch(1)\n",
+        "ds"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "Su_P8s8ZA41P",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "timeit(ds)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "HsLlXMO7EWBR",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### Cache"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "lV1NOn2zE2lR",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Use `tf.data.Dataset.cache` to easily take advantage of the performance boost of working with in memory data.\n",
+        "\n",
+        "Here the images are cached, after being pre-precessed (decoded and resized):"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "qj_U09xpDvOg",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "ds = image_label_ds.cache().repeat()\n",
+        "ds = ds.shuffle(buffer_size=4000) \n",
+        "ds = ds.batch(BATCH_SIZE).prefetch(1)\n",
+        "ds"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "rdxpvQ7VEo3y",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "timeit(ds)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "usIv7MqqZQps",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "One disadvantage to using an im memory cache is that the cache must be rebuilt on each run, giving the same startup delay each time the dataset is started:"
```

Comment:
I think @rohan100jain has some changes in flight that will help this no longer be the case.

Reply:
!

How does that work?